frontend/app/components/wp-relations/wp-relations.directive.ts

```diff
@@ -26,123 +26,147 @@
 // See doc/COPYRIGHT.rdoc for more details.
 //++
 
-import {wpTabsModule} from '../../angular-modules';
-import {WorkPackageRelationGroup} from './wp-relation-group/wp-relation-group.service';
-import {WorkPackageNotificationService} from '../wp-edit/wp-notification.service';
-import {WorkPackageResource} from '../api/api-v3/hal-resources/work-package-resource.service';
+import {wpDirectivesModule} from "../../angular-modules";
+import {RelatedWorkPackage, RelatedWorkPackagesGroup} from './wp-relations.interfaces';
 
-const iconArrowClasses = ['icon-arrow-up1', 'icon-arrow-down1'];
+import {WorkPackageResourceInterface} from '../api/api-v3/hal-resources/work-package-resource.service';
+import {WorkPackageCacheService} from '../work-packages/work-package-cache.service';
 
 export class WorkPackageRelationsController {
-  public btnTitle:string;
-  public btnIcon:string = '<i class="icon-hierarchy icon-add"></i>';
-
-  public workPackage:WorkPackageResource;
-  public relationGroup:WorkPackageRelationGroup;
-  public focusElementIndex:number = -2;
-  public wpToAddId:number = null;
-  public expand:boolean = false;
-  public text:any;
-
-  private _initialExpand:boolean;
-
-  public get stateClass():string {
-    return iconArrowClasses[+!!this.expand];
+  public relationGroups:RelatedWorkPackagesGroup;
+  public workPackage:WorkPackageResourceInterface;
+
+  public currentRelations: Array<RelatedWorkPackage> = Array();
```

Comment:
Holy fuck mate. This line though.
Use  `public currentRelations: RelatedWorkPackage[] = [];` syntax.
Your annotation is way too verbose and `Array` is not supposed to be used like that.
For new arrays, ALWAYS use `[]`